pkg/serializer/split/split.go

```diff
@@ -94,6 +96,7 @@ func Payloads(m marshaler.Marshaler, compress bool, mType MarshalType) (forwarde
 			chunks, err := toSplit.SplitPayload(numChunks)
 			log.Debugf("payload was split into %f chunks", len(chunks))
 			if err != nil {
+				handleDroppedPayloads(tempSlice[i:], marshallers)
```

Comment:
We iterate through `tempSlice`, when we succesfully split a payload it is appended to `smallEnoughPayloads` if it is small enough and to `marshallers` otherwise. When an error occurs at step `i` all the previous payloads are in these to slices. The dropped payloads are then in `marshallers` and the end of `tempSlice`.